File: golang/prime.go
package main

import (
	"fmt"
	"time"
	"os"
	"strconv"
)

func main() {
	limit := 10000
	if len(os.Args) == 2 {
		limit, _ = strconv.Atoi(os.Args[1]);
	}

	start := time.Now()
	count := 0
	for n := 2; ; n++ {
		if !isPrime(n) {
			continue
		}
		count++
		if count == limit {
			fmt.Printf("%dth prime is %d\n", limit, n)
			break
		}
	}
	duration := time.Now().Sub(start)
	fmt.Printf("[go] duration=%d msec\n", duration.Nanoseconds() / (1000*1000))
}

func isPrime(number int) bool {
	for n := 2; n <= number / 2; n++ {
		if number % n == 0 {
			return false
		}
	}
	return true
}

File: java/Prime.java

public class Prime
{
	public static void main(String[] args)
	{
		int limit = 10000;
		if(args.length == 1) limit = Integer.parseInt(args[0]);

		long start = System.currentTimeMillis();
		int count = 0;
		for(int number = 2; ; number++)
		{
			if(!isPrime(number)) continue;
			count++;
			if(count == limit)
			{
				System.out.println(limit + "th prime number is " + number);
				break;
			}
		}
		System.out.println("[java] duration is " + (System.currentTimeMillis() - start) + " ms");
	}

	private static boolean isPrime(int number)
	{
		for(int n = 2; n <= number/2; n++)
			if(number % n == 0) return false;
		return true;
	}
}
